### PR TITLE
HPCC-14735 Add session based authentication to ESP

### PR DESCRIPTION
**Pull request by wangkx**
ESP session Information is stored into dali. Also add
default login/logout html page.

Signed-off-by: wangkx kevin.wang@lexisnexis.com

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck I create this PR as we discussed. Please take a look...

File: esp/files/userlogon.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            function onLoad()
+            {
+                document.getElementById("username").focus();
+            }
+         </script>
+    </head>
+    <body class="yui-skin-sam" onload="onLoad()">
+        <p align="left" />
+        <h3>Please log on.</h3>
+        <form id="user_input_form" name="user_input_form" method="GET" action="/">
```
Comment (afishbeck) [inline esp/files/userlogon.html]: Having this method="GET" will cause the username and password to show in the browser URL.

**Comment by afishbeck** [comment]
@wangkx I think this mostly looks ok, but can you send me a config (offline) I can use to hit the ldap server and step through this?  It's a bit complex to just look at the code.

**Comment by afishbeck** [comment]
@wangkx have you verified that WsSecurity credentials still works?

I think it used to go through a different code path than basic auth credentials.

That is where instead of passing in basic authentication, the credentials are passed in a section of the SOAP header:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" xmlns="urn:hpccsystems:ws:wsworkunits">
 <soap:Header>	     
  <wsse:Security xmlns:wsse="http://schemas.xmlsoap.org/ws/2003/06/secext">
   <wsse:UsernameToken>
    <wsse:Username>myname</wsse:Username>
    <wsse:Password>mypassword</wsse:Password>
   </wsse:UsernameToken>
  </wsse:Security>
 </soap:Header>
 <soap:Body>
  <WUQueryRequest>
   <PageSize>0</PageSize>
   <PageStartFrom>0</PageStartFrom>
   <Descending>0</Descending>
  </WUQueryRequest>
 </soap:Body>
</soap:Envelope>
```

**Comment by richardkchapman** [comment]
@wangkx This seems to have stalled?

**Pull request closed without merging**